src/payment/strategies/moneris/moneris-payment-strategy.ts

```diff
@@ -89,29 +141,72 @@ export default class MonerisPaymentStrategy implements PaymentStrategy {
-                paymentData: { nonce },
+                paymentData: { nonce, shouldSaveInstrument, shouldSetAsDefaultInstrument },
             }));
         }
 
         return this._store.getState();
     }
 
-    finalize(): Promise<InternalCheckoutSelectors> {
-        return Promise.reject(new OrderFinalizationNotRequiredError());
-    }
+    private async _executeWithVaulted(payment: OrderPaymentRequestBody): Promise<InternalCheckoutSelectors> {
+        if (this._isHostedPaymentFormEnabled(payment.methodId)) {
+            const form = this._hostedForm;
 
-    deinitialize(): Promise<InternalCheckoutSelectors> {
-        if (this._windowEventListener) {
-            window.removeEventListener('message', this._windowEventListener);
-            this._windowEventListener = undefined;
+            if (!form) {
+                throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+            }
+
+            await form.validate();
+            await form.submit(payment);
+
+            return await this._store.dispatch(this._orderActionCreator.loadCurrentOrder());
+        } else {
+            return await this._store.dispatch(this._paymentActionCreator.submitPayment(payment));
         }
+    }
 
-        if (this._iframe && this._iframe.parentNode) {
-            this._iframe.parentNode.removeChild(this._iframe);
-            this._iframe = undefined;
+    private _shouldShowTSVHostedForm(methodId: string): boolean {
+        return (this._isHostedPaymentFormEnabled(methodId) && this._isHostedFieldAvailable());
+    }
+
+    private _isHostedPaymentFormEnabled(methodId: string): boolean {
+        const { paymentMethods: { getPaymentMethodOrThrow } } = this._store.getState();
+        const paymentMethod = getPaymentMethodOrThrow(methodId);
+
+        return paymentMethod.config.isHostedFormEnabled === true;
+    }
+
+    private _isHostedFieldAvailable(): boolean {
+        const options = this._getInitializeOptions();
+
+        return !!options.form?.fields;
+    }
+
+    private _getInitializeOptions(): MonerisPaymentInitializeOptions {
+        if (!this._initializeOptions) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
         }
 
-        return Promise.resolve(this._store.getState());
+        return this._initializeOptions;
+    }
+
+    private async _mountCardVerificationfields(formOptions: HostedFormOptions): Promise<HostedForm> {
+        try {
+            const { config } = this._store.getState();
+            const bigpayBaseUrl = config.getStoreConfig()?.paymentSettings.bigpayBaseUrl;
+
+            if (!bigpayBaseUrl) {
+                throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+            }
+
+            const form = this._hostedFormFactory.create(bigpayBaseUrl, formOptions);
+
+            await form.attach();
+
+            return form;
+        } catch (error) {
+            return error;
```

Comment:
This `error` is not of type `HostedForm` - Typescript is being fooled by it being an `any`.

This could lead to use accidentally trying to execute `HostedForm` methods on what is really an error object: https://github.com/bigcommerce/checkout-sdk-js/pull/1203/files#diff-d755dc9987b2df0ced566e63258108d7af6892bc592ed1cccd9f4ddbe119dfc9R159

Do we want to throw this error rather than return it? In which case we don't need the `try/catch` block.